tests/api/activitypub/signed_request.test.js

```diff
@@ -0,0 +1,160 @@
+const _ = require('builders/utils')
+const CONFIG = require('config')
+require('should')
+const { createUsername, createUserOnFediverse } = require('../fixtures/users')
+const { startServerWithEmitterAndReceiver, startServerWithEmitterUser, createReceiver, makeUrl } = require('../utils/activitypub')
+const { rawRequest } = require('../utils/request')
+const { shouldNotBeCalled, rethrowShouldNotBeCalledErrors, signedReq } = require('../utils/utils')
+const { sign } = require('controllers/activitypub/lib/security')
+
+const endpoint = '/api/activitypub'
+
+describe('activitypub:signed:request', () => {
+  it('should reject missing params', async () => {
+    try {
+      await rawRequest('get', `${endpoint}?action=actor`, {
+        headers: {
+          'content-type': 'application/activity+json'
+        }
+      })
+      .then(shouldNotBeCalled)
+    } catch (err) {
+      rethrowShouldNotBeCalledErrors(err)
+      const parsedBody = JSON.parse(err.body)
+      parsedBody.status.should.equal(400)
+      parsedBody.status_verbose.should.startWith('missing parameter')
+    }
+  })
+
+  it('should reject unsigned request', async () => {
+    try {
+      const receiverUsername = createUsername()
+      await rawRequest('get', `${endpoint}?action=actor&name=${receiverUsername}`, {
+        headers: {
+          'content-type': 'application/activity+json'
+        }
+      })
+      .then(shouldNotBeCalled)
+    } catch (err) {
+      rethrowShouldNotBeCalledErrors(err)
+      const parsedBody = JSON.parse(err.body)
+      parsedBody.status.should.equal(400)
+      parsedBody.status_verbose.should.equal('no signature header')
+    }
+  })
+
+  it('should reject when no publicKey is found', async () => {
+    try {
+      const emitterUser = await createUserOnFediverse()
```

Comment:
It's a bit confusing that the `emitterUser` supposedly from another instance, is actually created in the local instance database. It would be clearer if `tests/api/utils/activitypub.js` was just using directly `createKeyPair`, relying on simply in-memory user data